src/components/SearchForm.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { Field, Form, withFormik } from 'formik';
+import * as Yup from 'yup';
+
+const SearchForm = ({
+  values,
+  errors,
+  touched,
+}) => (
+  <Form>
+    <div className="form-field">
+      <label>Include these words or phrases
+        <Field
+          type="text"
+          name="searchTerms"
+          placeholder="seperate search terms with a comma (,)"
+        />
+      </label>
+      {touched.searchTerms && errors.searchTerms &&
+      <p className="form-error">{errors.searchTerms}</p>}
+    </div>
+    <div className="form-field">
+      <label>Location (optional, zipcode)
+        <Field
+          type="text"
+          name="location"
+          placeholder="97015"
+        />
+      </label>
+      {touched.location && errors.location &&
+      <p className="form-error">{errors.location}</p>}
+    </div>
+    <div className="form-field">
+      <label>radius (in miles)
+        <Field
+          type="text"
+          name="radius"
+          placeholder="25"
+        />
+      </label>
+      {touched.location && errors.radius &&
+      <p className="form-error">{errors.radius}</p>}
+    </div>
+    <div className="form-field">
+      <label>Timeframe (optional)
+        <Field
+          type="text"
+          name="timeFrame"
+          placeholder="1 month"
+        />
+      </label>
+      {touched.timeFrame && errors.timeFrame &&
+      <p className="form-error">{errors.timeFrame}</p>}
+    </div>
+    <label>
+      <Field
+        type="checkbox" name="savedSearch" checked={values.savedSearch} />
+      Save this search?
+    </label>
+    <br/>
+    <Field component="select" name="plan">
+      <option value="free">Free</option>
+      <option value="premium">Premium</option>
+    </Field>
+    <br/>
+    <button type="submit">Submit</button>
+    <button
+      type="button"
+      onClick={() => console.log('clicked clear')}
+    >Clear</button>
+  <button
+    type="button"
+    onClick={() => console.log('clicked "Save Search"')}
+  >Save Search</button>
+  </Form>
+)
+
+
+export const FormikSearch = withFormik({
+  mapPropsToValues({
```

Comment:
Does this work?

I'm not familiar with this syntax.  It seems like this could be
```
mapPropsToValues: ({...}) => {...}
```

But I'm not sure what is going on with this mapPropsToValues method.

Reply:
This is really cool!  I didn't know you could declare methods on objects this way!